src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx

```diff
@@ -1,66 +1,67 @@
+import type {CustomRendererProps, TPhrasing, TText} from 'react-native-render-html';
 import {TNodeChildrenRenderer} from 'react-native-render-html';
-import _ from 'underscore';
 import {usePersonalDetails} from '@components/OnyxProvider';
 import {ShowContextMenuContext, showContextMenuForReport} from '@components/ShowContextMenuContext';
 import Text from '@components/Text';
 import UserDetailsTooltip from '@components/UserDetailsTooltip';
 import withCurrentUserPersonalDetails from '@components/withCurrentUserPersonalDetails';
+import type {WithCurrentUserPersonalDetailsProps} from '@components/withCurrentUserPersonalDetails';
 import useLocalize from '@hooks/useLocalize';
 import useStyleUtils from '@hooks/useStyleUtils';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as LocalePhoneNumber from '@libs/LocalePhoneNumber';
 import Navigation from '@libs/Navigation/Navigation';
 import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
 import * as ReportUtils from '@libs/ReportUtils';
-import personalDetailsPropType from '@pages/personalDetailsPropType';
 import CONST from '@src/CONST';
 import ROUTES from '@src/ROUTES';
-import htmlRendererPropTypes from './htmlRendererPropTypes';
+import type {Route} from '@src/ROUTES';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 
-const propTypes = {
-    ...htmlRendererPropTypes,
+type MentionUserRendererProps = WithCurrentUserPersonalDetailsProps & CustomRendererProps<TText | TPhrasing>;
 
-    /** Current user personal details */
-    currentUserPersonalDetails: personalDetailsPropType.isRequired,
-};
-
-function MentionUserRenderer(props) {
+function MentionUserRenderer({style, tnode, TDefaultRenderer, currentUserPersonalDetails, ...defaultRendererProps}: MentionUserRendererProps) {
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const {translate} = useLocalize();
-    const defaultRendererProps = _.omit(props, ['TDefaultRenderer', 'style']);
-    const htmlAttribAccountID = lodashGet(props.tnode.attributes, 'accountid');
+    const htmlAttribAccountID = tnode.attributes.accountid;
     const personalDetails = usePersonalDetails() || CONST.EMPTY_OBJECT;
 
-    let accountID;
-    let displayNameOrLogin;
-    let navigationRoute;
+    let accountID: number;
+    let displayNameOrLogin: string;
+    let navigationRoute: Route;
 
-    if (!_.isEmpty(htmlAttribAccountID)) {
-        const user = lodashGet(personalDetails, htmlAttribAccountID);
+    if (!isEmpty(htmlAttribAccountID)) {
+        const user = personalDetails.htmlAttribAccountID;
         accountID = parseInt(htmlAttribAccountID, 10);
-        displayNameOrLogin = LocalePhoneNumber.formatPhoneNumber(lodashGet(user, 'login', '')) || lodashGet(user, 'displayName', '') || translate('common.hidden');
+        displayNameOrLogin = LocalePhoneNumber.formatPhoneNumber(user?.login ?? '') ?? user?.displayName ?? translate('common.hidden');
```

Comment:
You can disable eslint rule in this case to escape regression
```suggestion
        displayNameOrLogin = LocalePhoneNumber.formatPhoneNumber(user?.login ?? '') || user?.displayName || translate('common.hidden');
```